app/src/main/java/io/ushakov/bike_workouts/data_engine/WorkoutDataProcessor.kt

```diff
@@ -150,57 +151,158 @@ class WorkoutDataProcessor(
                     kiloCalories = workoutCalories
                 ))
 
-            Log.d("DBG", "Workout created with ${workoutId} id")
+            //Duration will be added into DB now and we will update this duration when workout pause
+            //On resume, we will create a new duration and will update on pause
+            //TODO create separate suspend function
+            activeDuration = Duration(0, workoutId, workout.startAt, null)
+            val durationId = durationRepository.insert(activeDuration!!)
+            activeDuration!!.id = durationId
+            //createNewDuration()
+            //Log.d("DBG", "Duration created with $durationId id")
+            //Log.d("DBG", "Workout created with $workoutId id")
         }
     }
 
     fun pauseWorkout() {
         val workout = activeWorkout ?: return
+        val onGoingDuration = activeDuration ?: return
+
 
-        calculateCalories(Date())
 
         coroutineScope.launch(Dispatchers.IO) {
             workoutRepository.setWorkoutStatus(workout.id, false)
 
 
+            //TODO remove it after saving into state or observable whatever.
+            //get active duration from DB
+            /*val getDurationJob = async {
+                durationRepository.getById(activeWorkout!!.id)
+            }*/
+
+            //update it
+            //TODO Use following line after saving durationToUpdate into state
+            //activeDuration?.stopAt = Date()
+            //val durationToUpdate = getDurationJob.await()
+            //durationToUpdate.stopAt = Date()
+            onGoingDuration.stopAt = Date()
+            activeDuration!!.stopAt = Date()
+            //TODO remove async after feature is tested
+            val updateDurationJob = async { durationRepository.update(onGoingDuration) }
```

Comment:
```suggestion
            val updateDurationJob = durationRepository.update(onGoingDuration)
```
no need for the async part